src/server/status/cgroup.js

```diff
@@ -0,0 +1,138 @@
+      const response = {};
+
+      data.toString().split(/\n/).forEach(line => {
+        const matches = line.match(CONTROL_GROUP_RE);
+
+        if (matches === null) {
+          return;
+        }
+
+        const controllers = matches[1].split(CONTROLLER_SEPERATOR_RE);
+        controllers.forEach(controller => {
+          response[controller] = matches[2];
+        });
+      });
+
+      return response;
+    });
+}
+
+function fileContentsToInteger(path) {
+  return readFile(path).then(data => {
+    return parseInt(data.toString());
+  });
+}
+
+function readCPUAcctUsage(controlGroup) {
+  return fileContentsToInteger(joinPath(PROC_CGROUP_CPUACCT_DIR, controlGroup, CPUACCT_USAGE_FILE));
+}
+
+function readCPUFsPeriod(controlGroup) {
+  return fileContentsToInteger(joinPath(PROC_CGROUP_CPU_DIR, controlGroup, CPU_FS_PERIOD_US_FILE));
+}
+
+function readCPUFsQuota(controlGroup) {
+  return fileContentsToInteger(joinPath(PROC_CGROUP_CPU_DIR, controlGroup, CPU_FS_QUOTA_US_FILE));
+}
+
+export function readCPUStat(controlGroup) {
+  return new Promise((resolve, reject) => {
+    const stat = {
+      number_of_elapsed_periods: -1,
+      number_of_times_throttled: -1,
+      time_throttled_nanos: -1
+    };
+
+    readFile(joinPath(PROC_CGROUP_CPU_DIR, controlGroup, CPU_STATS_FILE)).then(data => {
+      data.toString().split(/\n/).forEach(line => {
+        const fields = line.split(/\s+/);
+
+        switch(fields[0]) {
+          case 'nr_periods':
+            stat.number_of_elapsed_periods = parseInt(fields[1]);
+            break;
+
+          case 'nr_throttled':
+            stat.number_of_times_throttled = parseInt(fields[1]);
+            break;
+
+          case 'throttled_time':
+            stat.time_throttled_nanos = parseInt(fields[1]);
+            break;
+        }
+      });
+
+      resolve(stat);
+    }).catch(err => {
+      if (err.code === 'ENOENT') {
+        return resolve(stat);
+      }
+
+      reject(err);
+    });
+  });
+}
+
+export function getAllStats(options = {}) {
+  return readControlGroups().then(groups => {
+    const cpuPath = options.cpuPath || groups[GROUP_CPU];
+    const cpuAcctPath = options.cpuAcctPath || groups[GROUP_CPUACCT];
+
+    return Promise.all([
+      readCPUAcctUsage(cpuAcctPath),
+      readCPUFsPeriod(cpuPath),
+      readCPUFsQuota(cpuPath),
+      readCPUStat(cpuPath)
+    ]).then(([ cpuAcctUsage, cpuFsPeriod, cpuFsQuota, cpuStat ]) => {
+      const stats = {
+        cpuacct: {
+          control_group: cpuAcctPath,
+          usage_nanos: cpuAcctUsage
+        },
+
+        cpu: {
+          control_group: cpuPath,
+          cfs_period_micros: cpuFsPeriod,
+          cfs_quota_micros: cpuFsQuota,
+          stat: cpuStat
+        }
+      };
+
+      return stats;
+    }).catch(throwUnlessFileNotFound);
+  }).catch(throwUnlessFileNotFound);
+
+  function throwUnlessFileNotFound(err) {
+    if (err.code === 'ENOENT') {
+      return null;
+    }
+
+    throw err;
```

Comment:
how would you feel about logging this error instead of throwing?  if something comes up we can still return non c-group stats